### PR TITLE
feat(console): user role select

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

#582 has been merged into wrong branch, so reopen now.

sorry for the accident.

@simeng-li @demonzoo

### CODE REVIEW

File: packages/core/src/routes/admin-user.ts [hunk 1]
```diff
@@ -130,6 +131,20 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
...
+        const roles = await findRolesByRoleNames(roleNames);
+
+        if (roles.length !== roleNames.length) {
+          const resourcesNotFound = roleNames.filter(
+            (roleName) => !roles.some(({ name }) => roleName === name)
+          );
+          // TODO: Should be cached by the error handler and return request error
+          throw new InvalidInputError(`role names (${resourcesNotFound.join(',')}) are not valid`);
+        }
+      }
```
Comment (IceHe) [inline packages/core/src/routes/admin-user.ts:146]: nit: better compute the intersection of `roleNames` and `roles.name`s

Comment (wangsijie) [reply packages/core/src/routes/admin-user.ts:146]: This is old code.

**Review by demonzoo — APPROVED** [review]

File: packages/core/src/routes/admin-user.ts (hunk 1 above)
Comment (IceHe) [reply packages/core/src/routes/admin-user.ts:146]: There is a bug.
If lengths are the same and the content is different, it will not throw.

Comment (wangsijie) [reply packages/core/src/routes/admin-user.ts:146]: YES, but this is a "Temp solution" (see the first line of comment)

**Pull request merged**